app/controllers/incomeSources/add/AddIncomeSourceStartDateCheckController.scala

```diff
@@ -177,39 +175,55 @@ class AddIncomeSourceStartDateCheckController @Inject()(authenticate: Authentica
                               successUrl: String,
                               incomeSourceStartDate: String,
                               incomeSourceType: IncomeSourceType)
-                             (implicit mtdItUser: MtdItUser[_]): Result = {
+                             (implicit mtdItUser: MtdItUser[_]): Future[Result] = {
 
     val formResponse: Option[String] = validForm.toFormMap(form.response).headOption
 
     (formResponse, incomeSourceType) match {
       case (Some(form.responseNo), _) =>
-        Redirect(backUrl)
-          .removingFromSession(
-            incomeSourceType.startDateSessionKey
-          )
+        removeSessionData(Seq(incomeSourceType.startDateSessionKey), Redirect(backUrl), showInternalServerError(isAgent))
       case (Some(form.responseYes), SelfEmployment) =>
-        Redirect(successUrl)
-          .addingToSession(
-            SessionKeys.addBusinessAccountingPeriodStartDate -> incomeSourceStartDate,
-            SessionKeys.addBusinessAccountingPeriodEndDate -> dateService.getAccountingPeriodEndDate(incomeSourceStartDate)
-          )
+        setSessionData(addBusinessAccountingPeriodStartDate, incomeSourceStartDate, Redirect(successUrl)) flatMap {
```

Comment:
we would need to combine setting session multiple values under one methods in one go, so this would be replaced eventually by relevant method in SessionService